cnf-tests/testsuites/pkg/machineconfigpool/machineconfigpool.go

```diff
@@ -129,6 +130,50 @@ func WaitForMCPUpdated(mcp mcov1.MachineConfigPool) error {
 		time.Duration(30*mcp.Status.MachineCount)*time.Minute)
 }
 
+// WaitForAtLeastOneMCPUpdating waits until at least one MachineConfigPools in the cluster is updating
+func WaitForAtLeastOneMCPUpdating() error {
+	return wait.PollImmediate(3*time.Second, 2*time.Minute, func() (bool, error) {
+		mcpList, err := testclient.Client.MachineConfigPools().List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return false, nil
+		}
+
+		for _, mcp := range mcpList.Items {
+			if mcpHasCondition(&mcp, mcov1.MachineConfigPoolUpdating, corev1.ConditionTrue) {
+				klog.Infof("MCP %s is updating", mcp.Name)
+				return true, nil
+			}
+		}
```

Comment:
Do we not use the `WaitForCondition` helper function here on purpose?

Reply:
Unfortunately, we can't. `WaitForCondition()` waits on a single MCP to meet a condition. This function loops over all MCPs, jumping to the next one if the condition is not met.

If you have an idea on how to refactor this, please share it.